api/v1alpha3/azurecluster_conversion.go

```diff
@@ -85,6 +96,19 @@ func (dst *AzureCluster) ConvertFrom(srcRaw conversion.Hub) error { // nolint
 		return err
 	}
 
+	if src.Spec.BastionSpec.AzureBastion != nil {
```

Comment:
I believe this should be marshaled as part of the `utilconversion.MarshalData` can and be available for restoration via `restored` above. I don't thing this block is needed.

Reply:
Oh, thanks for opening my eyes. I struggled to make the conversion test to pass and it was so easy after all.
Should hopefully be ok now.